rasterio/_warp.pyx

```diff
@@ -752,6 +752,7 @@ cdef class WarpedVRTReaderBase(DatasetReaderBase):
                 with nogil:
                     hds_warped = GDALCreateWarpedVRT(
                         hds, c_width, c_height, dst_gt, psWOptions)
+                    GDALSetProjection(hds_warped, dst_crs_wkt)
```

Comment:
Here is the bug fix.  AFAIK this is a valid thing to do in this context.

Reply:
Yeah, I was in this file for another issue last night and noticed that this was missing. I can't wait to see if this fixes my problem.

Reply:
@sgillies If you're working on #1266, it doesn't.  But I think I figured out at least the cause of that problem.

Reply:
Oh yeah? I felt like I was getting close yesterday but then exhaustion hit. Can you fill me in?

My VRT creation code is pretty sad, I'd love to clean it up a bit while I'm in there.

Reply:
@sgillies See https://github.com/mapbox/rasterio/issues/1266#issuecomment-367222609, although I'm still adding a few details.